### PR TITLE
ENG-917 Implement rich error message on the frontend

### PR DESCRIPTION
**Pull request by antonio-lazarenni**
This pull request adds a new script called "get-errors" to the package.json file, which downloads the errors.json file from the stakekit/i18n repository and converts it into a TypeScript file. It also adds a new component called ReachErrorModal, which displays error messages based on the errors.json file.

### CODE REVIEW

File: src/hooks/use-reach-errors.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+export const reachErrorKey = signal<ReachError | undefined>(undefined);
+
+interface ReachError {
+  message: ErrorsSet;
+  details?: { [key: string]: any };
+}
+
+APIManager.getInstance()!.interceptors.response.use(undefined, (error) => {
+  reachErrorKey.value = error.response.data as ReachError;
```
Comment (petar-omni) [inline src/hooks/use-reach-errors.ts:14]: we should add type guard to check if this is in fact `ReactError` and not some other http error data

File: src/hooks/use-reach-errors.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { APIManager } from "@stakekit/api-hooks";
+import { signal, useSignalEffect } from "@preact/signals-react";
+import { useState } from "react";
+import { ErrorsSet, errorsSet } from "../utils/errors";
+
+export const reachErrorKey = signal<ReachError | undefined>(undefined);
```
Comment (petar-omni) [inline src/hooks/use-reach-errors.ts:6]: Isn't this the same thing as storing state in context and exposing it globally, or signal provides some other benefits here I'm not aware? If global scope is needed, lets use context for this

**Review by petar-omni — COMMENTED** [review]
Good stuff! Left some minor comments

File: src/hooks/use-reach-errors.ts (hunk 2 above)
Comment (antonio-lazarenni) [reply src/hooks/use-reach-errors.ts]: You are right, the first version of the code used signal directly without any react hooks, but after introducing a useReachError hook as a wrapper, the Signal stopped making any sense.

**Review by petar-omni — APPROVED** [review]

**Pull request merged**